### PR TITLE
integration tests: re-enable set_node_on_first_message_only

### PR DESCRIPTION
**Pull request by tbarrella**
Commit Message:

integration tests: re-enable set_node_on_first_message_only

A series of changes and reverts caused this to seemingly unintentionally stay disabled

Signed-off-by: Taylor Barrella <tabarr@google.com>

Additional Description:
Risk Level: Low
Testing: Integration tests
Docs Changes: N/A
Release Notes: N/A

### CODE REVIEW

File: test/config/utility.cc [hunk 1]
```diff
@@ -265,7 +263,7 @@ std::string ConfigHelper::discoveredClustersBootstrap(const std::string& api_typ
       grpc_services:
         envoy_grpc:
           cluster_name: my_cds_cluster
-      set_node_on_first_message_only: false
+      set_node_on_first_message_only: true
```
Comment (htuch) [inline test/config/utility.cc:266]: If this is default false, why not leave the integration tests also default false and just have some dedicated tests for it?

Comment (tbarrella) [reply test/config/utility.cc:266]: This is just trying to revert to how things were before #7293, #8478, and #9044. The first added the TODOs/switched true to false, the second removed the TODOs/switched false to true, and the third was a partial revert

Comment (htuch) [reply test/config/utility.cc:266]: OK. Do we have any tests that verify flows when it is false?

Comment (tbarrella) [reply test/config/utility.cc:266]: Looking at #7876, I don't think so, because `expect_node = false` by default

Comment (htuch) [reply test/config/utility.cc:266]: Could you add at least one? Thanks.

**Comment by tbarrella** [comment]
@snowp do you have a chance to review this?

File: test/integration/ads_integration_test.cc [hunk 2]
```diff
@@ -1546,4 +1546,42 @@ TEST_P(AdsClusterV2Test, DEPRECATED_FEATURE_TEST(TypeUrlAnnotationRegression)) {
   test_server_->waitForCounterGe("cluster_manager.cds.update_rejected", 1);
 }
 
+class AdsSetNodeAlwaysIntegrationTest : public AdsIntegrationTest {
+public:
+  AdsSetNodeAlwaysIntegrationTest() = default;
```
Comment (snowp) [inline test/integration/ads_integration_test.cc:1551]: nit: this is not necessary, just remove this line

File: test/integration/ads_integration_test.cc [hunk 3]
```diff
@@ -1546,4 +1546,42 @@ TEST_P(AdsClusterV2Test, DEPRECATED_FEATURE_TEST(TypeUrlAnnotationRegression)) {
   test_server_->waitForCounterGe("cluster_manager.cds.update_rejected", 1);
 }
 
+class AdsSetNodeAlwaysIntegrationTest : public AdsIntegrationTest {
```
Comment (snowp) [inline test/integration/ads_integration_test.cc:1549]: This seems like a bit much boilerplate for not much gain, could we just write this test as regular test case within `AdsIntegrationTest` where we call `addConfigModifier` before `initialize`?

**Review by snowp — CHANGES_REQUESTED** [review]
Thanks this looks reasonable to me, just some questions around the test structure

File: test/integration/ads_integration_test.cc (hunk 2 above)
Comment (tbarrella) [reply test/integration/ads_integration_test.cc:1551]: thank you!

File: test/integration/ads_integration_test.cc (hunk 3 above)
Comment (tbarrella) [reply test/integration/ads_integration_test.cc:1549]: Sounds good to me

**Review by htuch — APPROVED** [review]
LGTM, thanks!

**Review by snowp — APPROVED** [review]
Thanks!

**Pull request merged**